packages/bigquery/src/index.js

```diff
@@ -77,6 +79,94 @@ function BigQuery(options) {
 util.inherits(BigQuery, common.Service);
 
 /**
+ * Merge a rowset returned from the API with a table schema.
+ *
+ * @private
+ *
+ * @param {object} schema
+ * @param {array} rows
+ * @return {array} Fields using their matching names from the table's schema.
+ */
+BigQuery.mergeSchemaWithRows_ =
+BigQuery.prototype.mergeSchemaWithRows_ = function(schema, rows) {
+  return arrify(rows).map(mergeSchema).map(flattenRows);
+
+  function mergeSchema(row) {
+    return row.f.map(function(field, index) {
+      var schemaField = schema.fields[index];
+      var value = field.v;
+
+      if (schemaField.mode === 'REPEATED') {
+        value = value.map(function(val) {
+          return convert(schemaField, val.v);
+        });
+      } else {
+        value = convert(schemaField, value);
+      }
+
+      var fieldObject = {};
+      fieldObject[schemaField.name] = value;
+      return fieldObject;
+    });
+  }
+
+  function convert(schemaField, value) {
+    if (is.nil(value)) {
+      return value;
+    }
+
+    switch (schemaField.type) {
+      case 'BOOLEAN': {
+        value = value === 'true';
+        break;
+      }
+      case 'BYTES': {
+        value = new Buffer(value, 'base64');
+        break;
+      }
+      case 'FLOAT': {
+        value = parseFloat(value);
+        break;
+      }
+      case 'INTEGER': {
```

Comment:
I think you might need two cases here eventually. Standard SQL renamed some of the types, such as `INT64` https://cloud.google.com/bigquery/docs/reference/standard-sql/data-types#integer-type